library/LuaApi.cpp

```diff
@@ -1350,8 +1355,8 @@ static string upperCp437(string s) { return toUpper_cp437(s); }
 static string lowerCp437(string s) { return toLower_cp437(s); }
 static string toSearchNormalized(string s) { return to_search_normalized(s); }
 static string capitalizeStringWords(string s) { return capitalize_string_words(s); }
-static string formatInt(int32_t num) { return format_number(num); }
-static string formatFloat(float num) { return format_number(num); }
+static string formatInt(int64_t num) { return format_number_by_locale(num, get_preferred_number_format_type()); }
+static string formatFloat(double num) { return format_number_by_locale(num, get_preferred_number_format_type()); }
```

Comment:
C++ has the template, so it doesn't need separate entrypoints like this. What I can do, though, is add a convenience wrapper that grabs the preference instead of making the client look it up themselves:
```cpp
template<typename T>
static inline std::string format_number_by_locale(T num) {
    format_number_by_locale(num, get_preferred_number_format_type());
}
```